Formula/ghc.rb

```diff
@@ -24,97 +19,63 @@ class Ghc < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "bea0a8011b453e37e99855a6640642303d60c6611db366cb5f2d02a7546da19b"
   end
 
-  depends_on "python@3.9" => :build
+  depends_on "gmp" => :build
```

Comment:
I think we can restore previous Linux-only rather than changes from `ghc@9` formula. We build with bundled `gmp` and pre-builts also have no linkage (links to `/usr/lib/libffi.dylib` and `/usr/lib/libncurses.5.4.dylib`) so don't know where macOS usage came from.

With new Ubuntu container, may not be needed on Linux as there is a system copy, but probably safer to keep for source-builds since those may be running older systems (but they may be broken due to how bootstraps are dynamically linked to system libraries).
```console
$ ldconfig -p | grep gmp
	libgmp.so.10 (libc6,x86-64) => /lib/x86_64-linux-gnu/libgmp.so.10
```